torch_geometric/nn/norm/batch_norm.py

```diff
@@ -30,18 +30,26 @@ class BatchNorm(torch.nn.Module):
             :obj:`False`, this module does not track such statistics and always
             uses batch statistics in both training and eval modes.
             (default: :obj:`True`)
+        skip_no_batch (bool, optional): If set to :obj:`True`, batches with
```

Comment:
I don't think this is a good choice since now we apply no normalization at all. Instead, IMO it is better to use training statistics for normalization.

Reply:
Thanks for the feedback. My only concern for this is that it may be based on a small subset unless we add to the running mean/variance even when the batch size is one. WDYT?

Reply:
Can you clarify? What's the problem with using running mean/variance here for normalization here?

Reply:
I've updated baed on your suggestion.

But to clarify: In the `BatchNorm` layer, we calculate mean and variance using a `running_mean` and `running_var` which are exponentially smoothed. But if we switch to `eval` mode for the cases where batch size is one, we will exclude those examples from the mean/var calculation.

I also suspect that with small batch sizes the variance/mean calculated this way will not approximate well the population.

Reply:
Yes, these examples will be excluded from the mean/var calculation, but I guess there is no way around this (it would be the case in your previous implementation as well). I think the longer we train, the more stable the running mean/var will become. Ideally, batches with only a single node for a node type should be not too frequent, so excluding them should not cause too much harm.

Reply:
> but I guess there is no way around this (it would be the case in your previous implementation as well). 

Yes agreed. The previous implementation was not better in this respect. It was just simpler and I thought I'd raise the question.